jhdf/src/main/java/io/jhdf/AbstractNode.java

```diff
@@ -145,7 +145,7 @@ public boolean isLink() {
 		return false;
 	}
 
-	protected <T extends Message> T getHeaderMessage(Class<T> clazz) {
+	public <T extends Message> T getHeaderMessage(Class<T> clazz) {
```

Comment:
Does this need to be public?

Reply:
The only thing that we are interested in is the address of a ContiguousDataset. With this change we can get it like this
`
ContiguousDataset contiguousDataset = (ContiguousDataset) dataset;
ContiguousDataLayoutMessage contiguousDataLayoutMessage = contiguousDataset.getHeaderMessage(ContiguousDataLayoutMessage.class);
contiguousDataLayoutMessage.getAddress();`
but we are open to other suggestions like an extra method for the address.